### PR TITLE
[#8082] Some dapps don't work in status when privacy mode is enabled

### LINKED ISSUE #8082: Some dapps don't work in status when privacy mode is enabled
# Problem

Some dapps don't support EIP 1193, and they just don't work in status when privacy mode is enabled

## Future Steps

Find out how we could inform users about this

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #8082

https://www.youtube.com/watch?v=YQWo-42uxQM

### CODE REVIEW

File: src/status_im/ui/screens/browser/open_dapp/views.cljs [hunk 1]
```diff
@@ -31,8 +34,39 @@
...
+(views/defview privacy-options []
+  (views/letsubs [{:keys [settings]} [:account/account]]
+    (let [privacy-enabled? (or (nil? (:web3-opt-in? settings)) (:web3-opt-in? settings))]
+      (when @privacy-otions-visible?
+        [react/view styles/privacy-container
+         [react/view {:style {:flex-direction :row}}
+          [react/view {:flex 1}
+           [react/text {:style {:typography :main-medium}} (i18n/label :t/dapps-can-access)]]
+          [react/touchable-highlight {:on-press #(reset! privacy-otions-visible? false)}
+           [react/view {:style {:width 40 :height 30 :align-items :center :margin-left 20}}
```
Comment (vitvly) [inline src/status_im/ui/screens/browser/open_dapp/views.cljs:48]: Can we move these styles to a separate ns?

**Review by vitvly — APPROVED** [review]

File: src/status_im/ui/screens/browser/open_dapp/views.cljs [hunk 2]
```diff
@@ -31,8 +34,39 @@
...
+
+(views/defview privacy-options []
+  (views/letsubs [{:keys [settings]} [:account/account]]
+    (let [privacy-enabled? (or (nil? (:web3-opt-in? settings)) (:web3-opt-in? settings))]
+      (when @privacy-otions-visible?
+        [react/view styles/privacy-container
+         [react/view {:style {:flex-direction :row}}
+          [react/view {:flex 1}
+           [react/text {:style {:typography :main-medium}} (i18n/label :t/dapps-can-access)]]
+          [react/touchable-highlight {:on-press #(reset! privacy-otions-visible? false)}
```
Comment (jeluard) [inline src/status_im/ui/screens/browser/open_dapp/views.cljs:47]: Global expected?

**Review by jeluard — APPROVED** [review]

**Comment by churik** [comment]
@flexsurfer `android-e2e` failed to build - tried 3 times:
```
2019-05-06T12:25:29.047Z] 

[2019-05-06T12:25:29.612Z] Could not locate Gemfile

script returned exit code 10
```
Probably rebase will help.

**Comment by flexsurfer** [comment]
thanks @churik , rebased

**Comment by asemiankevich** [comment]
@flexsurfer any examples what was not working?
also the youtube leads to nothing
image

also, second option is not visible (iPhone 7). + this tooltip is not scrollable. 
also, is it a duplicate of Profile -> Browser privacy mode?

image

**Comment by flexsurfer** [comment]
thanks, @asemiankevich good catches, idk video still works for me.
So we didn't fix something here, sorry about description, we just duplicated privacy option in the browser tab, and rephrased it, the main goal is to inform a user that permissions enabled may break some dapps
i'll fix issues thanks

**Comment by flexsurfer** [comment]
@asemiankevich fixed second option, removed privacy item from the profile, thanks

**Comment by asemiankevich** [comment]
so i checked:
- can enable permissions check -> then dapp is asking for permissions 
- if granting permissions -> can see them in profile -> dapp dapp permissions
- can revoke permissions -> force the dapp to ask them again by enabling the radiobutton
- can disable permissions check
- can close the hint window
- after i restart the app -> the hint is shown again

Checked on iPhone 7 and Android 8, looks good to me. Would be nice i think to automate this.

@Serhy would you mind to assist with checking autotests? this one looks scary

image

**Comment by Serhy** [comment]
@asemiankevich re-running failed tests now.

The error on screenshot is Status app bug. Can't reproduce with real device Android 8 on the same e2e build, but something separate we need to fix separately

From logcat its:
```

05-08 08:46:27.713  4559  4613 D ReactNativeJS: DEBUG [status-im.ui.screens.routing.core:31] - :on-will-focus :chat
05-08 08:46:27.716  4559  4613 D ReactNativeJS: DEBUG [status-im.utils.handlers:34] - Handling re-frame event:  :screens/on-will-focus
05-08 08:46:27.726  4559  4613 E ReactNativeJS: ERROR [status-im.models.transactions:99] - Request for block info failed
05-08 08:46:27.731  4559  4586 D EGL_emulation: eglMakeCurrent: 0xd1bd9ba0: ver 2 0 (tinfo 0xd1c7e160)
05-08 08:46:27.739  4559  4613 W ReactNativeJS: Unable to symbolicate stack trace: Bundle was not loaded from the packager
05-08 08:46:27.744  1410  1947 E SurfaceFlinger: ro.sf.lcd_density must be defined as a build property
05-08 08:46:27.745  4559  4613 D ReactNativeJS: DEBUG [status-im.ui.screens.views:73] - :main-component-did-update :chat

```

**Comment by asemiankevich** [comment]
okay great then will merge this one and log this error separately

**Pull request merged**